src/core/build.jl

```diff
@@ -12,9 +12,22 @@ function _instance_datatype(md::ModelDef, def::DatumDef, start::Int)
     elseif dims[1] != :time
         T = Array{dtype, num_dims}
     
-    else
-        step = step_size(md)
-        T = TimestepArray{dtype, num_dims, start, step}
+    else   
+        # OLD WAY
+        # step = step_size(md)
+        # T = TimestepArray{dtype, num_dims, start, step}
+
+        if isuniform(md)
+            #take the start from the function argument, not the model def
+            ~, stepsize = first_and_step(md)
```

Comment:
Don't assign something to ``~``, that symbol is already bound to a function. You can use ``_`` if you don't care about the first unpacked argument (i.e. ``_, stepsize = ...``).